bindings/rust/s2n-tls/src/raw/enums.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#![allow(clippy::missing_safety_doc)] // TODO add safety docs
+
+use crate::raw::error::Error;
+use core::convert::TryFrom;
+use s2n_tls_sys::*;
+
```

Comment:
`enum` feels odd just like a file called `struct` wouldn't feel right. Its too specific to a Rust language feature but instead it should be specific to the program. Tying the enums to connections still makes sense since the enums are related to the connection. So in order to de-clutter the file you could do the following:

- create a folder called `/connection` and move the file there (this is like a module in Rust)
- rename the file from `enum.rs` to `property.rs` or `configufation.rs` or a different name
- re-export the enums you want from connection.rs `pub use crate connection::property::*`

```
/src
  > connection.rs
  /connection
    > property.rs
...
```

Reply:
I considered re-exporting, but the problem is that s2n-tls doesn't make a distinction between connection enums and config enums. We use the same enums for both sets of APIs.

Reply:
There is also an entire section in our usage guide just called "enums" 😅 https://github.com/aws/s2n-tls/blob/main/docs/USAGE-GUIDE.md#enums That probably shouldn't exist because it's really not useful, but it does support enums being a distinct module.